client/src/main/java/org/apache/celeborn/client/read/PartitionReader.java

```diff
@@ -21,14 +21,21 @@
 
 import io.netty.buffer.ByteBuf;
 
+import org.apache.celeborn.client.read.checkpoint.PartitionReaderCheckpointMetadata;
 import org.apache.celeborn.common.protocol.PartitionLocation;
 
-public interface PartitionReader {
+public interface PartitionReader<T extends PartitionReaderCheckpointMetadata> {
   boolean hasNext();
 
   ByteBuf next() throws IOException, InterruptedException;
 
   void close();
 
   PartitionLocation getLocation();
+
+  default T getPartitionReaderCheckpointMetadata() {
```

Comment:
Why does this method has default implementation?

Reply:
I initially thought of implementing it only for the WorkerPartitionReader, but I can add no-op impl for other types of readers, and can take up their implementation in a subsequent PR